torch/testing/_internal/custom_op_db.py

```diff
@@ -83,9 +86,114 @@ def numpy_take_impl(x, ind, ind_inv, dim):
     ind = to_numpy(ind)
     return torch.tensor(np.take_along_axis(x, ind, dim), device=device)
 
-@numpy_take.impl_meta()
-def numpy_take_meta(x, ind, ind_inv, dim):
-    return x
+@numpy_take.impl_fake()
+def numpy_take_fake(x, ind, ind_inv, dim):
+    return torch.empty_like(x)
+
+@custom_op('(Tensor x) -> Tensor', ns='_torch_testing')
+def numpy_nonzero(x):
+    ...
+
+@numpy_nonzero.impl(['cpu', 'cuda'])
+def numpy_nonzero_impl(x):
+    x_np = to_numpy(x)
+    res = np.stack(np.nonzero(x_np), axis=1)
+    if res.shape[0] <= 1:
+        raise RuntimeError("not supported")
+    return torch.tensor(res, device=x.device)
+
+@numpy_nonzero.impl_fake()
+def numpy_nonzero_fake(x):
+    ctx = torch._custom_op.get_ctx()
+    i0 = ctx.new_data_dependent_symint()
+    shape = [x.dim(), i0]
+    result = x.new_empty(shape, dtype=torch.long)
+    return result
+
+def sample_inputs_numpy_nonzero(opinfo, device, dtype, requires_grad, **kwargs):
+    make_arg = functools.partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+    shape = 10
+    result = make_arg(shape, low=0.9, high=2)
+    mask = make_tensor(shape, low=0, high=2, device=device, dtype=torch.long)
+    with torch.no_grad():
+        result *= mask
+
+    yield SampleInput(result, args=())
+
+
+@custom_op('(Tensor boxes, Tensor scores, float iou_threshold) -> Tensor', ns='_torch_testing')
+def numpy_nms(boxes, scores, iou_threshold):
+    ...
+
+@numpy_nms.impl(['cpu', 'cuda'])
+def numpy_nms_impl(boxes, scores, iou_threshold):
+    # Adapted from Ross Girshick's fast-rcnn implementation at
+    # https://github.com/rbgirshick/fast-rcnn/blob/master/lib/utils/nms.py
+    assert boxes.device == scores.device
+    device = boxes.device
+
+    boxes = to_numpy(boxes)
+    scores = to_numpy(scores)
+
+    N = boxes.shape[0]
+    assert boxes.shape == (N, 4)
+    assert scores.shape == (N,)
+
+    x1 = boxes[:, 0]
+    y1 = boxes[:, 1]
+    x2 = boxes[:, 2]
+    y2 = boxes[:, 3]
+
+    areas = (x2 - x1 + 1) * (y2 - y1 + 1)
+    order = scores.argsort()[::-1]
+
+    keep = []
+    while order.size > 0:
+        i = order[0]
+        keep.append(i)
+        xx1 = np.maximum(x1[i], x1[order[1:]])
+        yy1 = np.maximum(y1[i], y1[order[1:]])
+        xx2 = np.minimum(x2[i], x2[order[1:]])
+        yy2 = np.minimum(y2[i], y2[order[1:]])
+
```

Comment:
How does this work? We're running this on meta tensors here, so I would think this would throw